wiki/osu!wiki_contribution_guide/en.md

```diff
@@ -0,0 +1,531 @@
+
+### Through GitHub's Web Editor
+
+#### Branches cleanups
+
+Go to your own forked repository.
+
+There are two ways of cleaning a branch, overview branch cleanup, and merged PR manual branch deletion.
+As the second way requires the branch to be merged for the branch delete button appear in the merged PR page, the first way of cleaning the branches is preferred for this guide.
+
+Click on the "Branches" icon and text to go to the branches page.
+
+![Branch page](Contribute_pr2.jpg "Branch Page")
+
+You should roughly get the screen as shown above.
+Click on the red trashcan icon on the branch you want to delete.
+The branch will be marked for deletion, and will be deleted after you leave the page.
+
+Once you have done cleaning the branches, it is time to update your own forked repository.
+
+_Note: For further reference, see GitHub Help answer: <https://help.github.com/articles/viewing-branches-in-your-repository/>_
+
+#### Branch sync
+
+Go to ``ppy/osu-wiki`` repository this time.
+
+_Note: While technically you can go back to your own repository to do the PR, the amount of clicks needed coupled by zealous page refresh the moment you change the fork or branch may soon become a test of your patience and your connection quota._
+
+![Main repo](Contribute_sync1.jpg "The master ppy's repo")
+
+Pick either one of the plan to access the PR compare page.
+
+If you go for plan B, click on the green "New pull request" button.
+
+![New pull request in main repo](Contribute_sync2.jpg "The PR page of ppy's repo")
+
+Now, you should see the page as shown below:
+
+![Comparing changes](Contribute_synccc.jpg "Comparing branch changes within same repo")
+
+Click on the blue keyword, _compare across forks_.
+The fork dropdown buttons will be added.
+Set the base fork as your own forked repository (the second one in the dropdown list, usually).
+A page refresh will occur when the fork was changed.
+
+You should see the page now as below:
+
+![Making a merge update PR](Contribute_syncpr.jpg "Making a self-repo PR update")
+
+Fill up the PR information (you can just write anything since it is your own repo), and press the green "Create pull request".
+
+You will be directed to your freshly-created PR page:
+
+![Preparing to merge](Contribute_syncmerge1.jpg "Preparing to merge")
+
+Scroll down to the lower part of the page.
+Always set your green "Merge pull request" button to "Create a merge commit", which is set by default.
+
+Press the green "Merge pull request" button.
+
+![Setting the merge message](Contribute_syncmerge2.jpg "Setting the merge message")
+
+Click on the green "Confirm merge" button, and the branch is updated successfully to the latest version.
+
+To update your other branches, do the same thing in this section in a new PR but with different branch this time.
+
+### Through your Local Machine
+
+#### Dealing with "vim"
+<!-- screenshot of vim -->
+
+_vim_ is Git's default terminal text-editor of choice by default.
```

Comment:
but on Linux(Ubuntu/Debian ....),sometimes Git's default terminal text-editor is `nano`